tests/cpp-runtime/hexagon/hexagon_user_dma_tests.cc

```diff
@@ -57,9 +57,9 @@ TEST_F(HexagonUserDMATest, wait) {
   HexagonUserDMA::Get().Wait(10);
 }
 
```

Comment:
Should we white-box test the ring-buffer impl?

Reply:
Added new RingBuffer class along with testing.